packages/es-client/tests/test-granule-upsert.js

```diff
@@ -452,12 +735,47 @@ test('upsertGranule does update "running" granule record to "failed" status for
     esClient,
     updates,
     index: esIndex,
+  }, true);
+  const updatedEsRecord = await t.context.esGranulesClient.get(granule.granuleId);
+  t.like(updatedEsRecord, updates);
+});
+
+test('upsertGranule does update "running" granule record to "failed" status for the same execution if writeConstraints is false', async (t) => {
+  const { esIndex, esClient } = t.context;
+
+  const execution = randomString();
+  const createdAt = Date.now();
+  const granule = {
+    granuleId: randomString(),
+    collectionId: randomString(),
+    status: 'running',
+    execution,
+    createdAt,
+  };
+
+  await indexer.upsertGranule({
+    esClient,
+    updates: granule,
+    index: esIndex,
   });
+
+  const esRecord = await t.context.esGranulesClient.get(granule.granuleId);
+  t.like(esRecord, granule);
+
+  const updates = {
+    ...granule,
+    status: 'failed',
+  };
+  await indexer.upsertGranule({
+    esClient,
+    updates,
+    index: esIndex,
+  }, false);
   const updatedEsRecord = await t.context.esGranulesClient.get(granule.granuleId);
   t.like(updatedEsRecord, updates);
```

Comment:
Perhaps check the status directly. Also unsure of the comparison here - should it be between updatedEsRecord and granule?

Reply:
This is another `does update` test -we expect this one would write.